cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -571,3 +589,252 @@ def _circuit_diagram_info_(
+        args = sim.ActOnCliffordTableauArgs(
+            tableau=t, qubits=qubits, prng=np.random.RandomState(), log_of_measurement_results={}
+        )
+
+        protocols.act_on(gate, args, qubits, allow_decompose=False)
+        return CliffordGate.from_clifford_tableau(args.tableau)
+
+
+def _pad_tableau(
+    clifford_tableau: qis.CliffordTableau, num_qubits_after_padding: int, axes: List[int]
+) -> qis.CliffordTableau:
+    """Roughly, this function copies self.tabluea into the "identity" matrix."""
+    # Sanity check
+    if len(set(axes)) != clifford_tableau.n:
+        raise ValueError(
+            "Input axes of padding should match with the number of qubits in the input tableau."
+        )
+    if clifford_tableau.n > num_qubits_after_padding:
+        raise ValueError(
+            "The number of qubits in the input tableau should not be larger than "
+            "num_qubits_after_padding."
+        )
+
+    padded_tableau = qis.CliffordTableau(num_qubits_after_padding)
+    v_index = np.concatenate((np.asarray(axes), num_qubits_after_padding + np.asarray(axes)))
+
+    padded_tableau.xs[np.ix_(v_index, axes)] = clifford_tableau.xs.copy()
+    padded_tableau.zs[np.ix_(v_index, axes)] = clifford_tableau.zs.copy()
+    padded_tableau.rs[v_index] = clifford_tableau.rs.copy()
+    return padded_tableau
+
+
+@value.value_equality
+class CliffordGate(raw_types.Gate, CommonCliffordGates):
+    """Clifford rotation for N-qubit."""
+
+    def __init__(
+        self,
+        *,
+        _clifford_tableau: qis.CliffordTableau,
+    ) -> None:
+        # We use the Clifford tableau to represent a Clifford gate.
+        # It is crucial to note that the meaning of tableau here is different
+        # from the one used to represent a Clifford state (Of course, they are related).
+        # A) We have to use the full 2n * (2n + 1) matrix
+        # B) The meaning of tableau here is
+        #                 X   Z    sign
+        #     from  X  [ X_x Z_x | r_x ]
+        #     from  Z  [ X_z Z_z | r_z ]
+        # Each row in the Clifford tableau means the transformation of original Pauli gates.
+        # For example, take a 2 * (2+1) tableau as example:
+        #         X       Z     r
+        #  XI  [ 1  0 | 1  0  | 0 ]
+        #  IX  [ 0  0 | 1  1  | 0 ]
+        #  ZI  [ 0  0 | 1  0  | 1 ]
+        #  IZ  [ 1  0 | 1  1  | 0 ]
+        # Take the third row as example: this means the ZI gate after the this gate,
+        # more precisely the conjugate transformation of ZI by this gate, becomes -ZI.
+        # (Note the real clifford tableau has to satify the Symplectic property.
+        # here is just for illustration)
+        self._clifford_tableau = _clifford_tableau.copy()
+
+    @property
+    def clifford_tableau(self):
+        return self._clifford_tableau
+
+    @classmethod
+    def from_clifford_tableau(cls, tableau: qis.CliffordTableau) -> 'CliffordGate':
+        assert isinstance(tableau, qis.CliffordTableau)
```

Comment:
Nit: Generally speaking we don't want to add type checking as a part of the API: https://google.github.io/styleguide/pyguide.html#doc-function-raises

Reply:
Docstring is also missing.

Reply:
Change to the raise exception and added the docstring